File: src/main/java/cat/mvmike/numbertotext/magnitude/Hundreds.java
package cat.mvmike.numbertotext.magnitude;

import cat.mvmike.numbertotext.language.Number;

import static cat.mvmike.numbertotext.language.Constants.DASH;
import static cat.mvmike.numbertotext.language.Constants.EMPTY;
import static cat.mvmike.numbertotext.language.Number.N_0;
import static cat.mvmike.numbertotext.language.Number.N_1;
import static cat.mvmike.numbertotext.language.Number.N_100;
import static cat.mvmike.numbertotext.language.Number.N_9;
import static cat.mvmike.numbertotext.language.Constants.PLURAL;
import static cat.mvmike.numbertotext.language.Number.getLiteralOpt;

public final class Hundreds extends BaseMagnitude {

    public Hundreds(int number) {
        super(number);
    }

    @Override
    public String get() {
        return getLiteralOpt(number, N_1, N_9)
                .map(Hundreds::toString)
                .orElse(EMPTY);
    }

    private static String toString(final Number literal) {
        switch (literal) {
            case N_1 -> { return N_100.getLiteral(); }
            case N_2, N_3, N_4, N_5, N_6, N_7, N_8, N_9 -> { return literal.getLiteral() + DASH + N_100.getLiteral() + PLURAL; }
            default -> throw new UnsupportedOperationException();
        }
    }

    @Override
    public boolean isMultipleOf() {
        return N_0.getValue() == (number % N_100.getValue());
    }
}

File: src/main/kotlin/cat/mvmike/numbertotext/magnitude/Hundreds.kt
// Copyright (c) 2016, Miquel Martí <miquelmarti111@gmail.com>
// See LICENSE for licensing information
package cat.mvmike.numbertotext.magnitude

import cat.mvmike.numbertotext.language.Constants.DASH
import cat.mvmike.numbertotext.language.Constants.EMPTY
import cat.mvmike.numbertotext.language.Constants.PLURAL
import cat.mvmike.numbertotext.language.NumberLiteral
import cat.mvmike.numbertotext.language.NumberLiteral.N_0
import cat.mvmike.numbertotext.language.NumberLiteral.N_1
import cat.mvmike.numbertotext.language.NumberLiteral.N_100
import cat.mvmike.numbertotext.language.NumberLiteral.N_2
import cat.mvmike.numbertotext.language.NumberLiteral.N_3
import cat.mvmike.numbertotext.language.NumberLiteral.N_4
import cat.mvmike.numbertotext.language.NumberLiteral.N_5
import cat.mvmike.numbertotext.language.NumberLiteral.N_6
import cat.mvmike.numbertotext.language.NumberLiteral.N_7
import cat.mvmike.numbertotext.language.NumberLiteral.N_8
import cat.mvmike.numbertotext.language.NumberLiteral.N_9
import cat.mvmike.numbertotext.language.getLiteral

class Hundreds(
    private val number: Int
) {

    fun get() = getLiteral(number, N_1, N_9)
        ?.toFormattedString()
        ?: EMPTY

    private fun NumberLiteral.toFormattedString() = when (this) {
        N_1 -> N_100.literal
        N_2,
        N_3,
        N_4,
        N_5,
        N_6,
        N_7,
        N_8,
        N_9 -> literal + DASH + N_100.literal + PLURAL
        else -> throw UnsupportedOperationException()
    }
}

fun Int.modOfHundred() = this % N_100.value

fun Int.isMultipleOfHundreds() = N_0.value == this.modOfHundred()